ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/drawing-and-display.tsx

```diff
@@ -61,6 +61,123 @@ export const removeOldDrawing = ({ map, id }: { map: any; id: string }) => {
   relevantPrimitives.length > 0 && map.getMap().scene.requestRender()
 }
 
+const makeOldDrawingNonEditable = ({ map, id }: { map: any; id: string }) => {
+  const relevantPrimitives = map
+    .getMap()
+    .scene.primitives._primitives.filter((prim: any) => {
+      return prim.id === id
+    })
+  relevantPrimitives.forEach((relevantPrimitive: any) => {
+    if (typeof relevantPrimitive.setEditMode === 'function') {
+      relevantPrimitive.setEditMode(false)
+    }
+    if (typeof relevantPrimitive.setEditable === 'function') {
+      relevantPrimitive.setEditable(false)
+    }
+  })
+  relevantPrimitives.length > 0 && map.getMap().scene.requestRender()
+}
+
+const nestedArraysOverlap = (arrayA: any[], arrayB: any[]) => {
+  let result = false
+  arrayA.map((elemA) => {
```

Comment:
Does this need to be a map function? The map will iterate over every element in arrayA, but it seems like it just cares that any element has overlap with any element of arrayB. I might not be understanding it correctly, but it seems like this would work?

```suggestion
  return arrayA.some((elemA) => arrayB.some((elemB) => JSON.stringify(elemA) === JSON.stringify(elemB)))
```

Reply:
I think that would work. Thanks!